grudge/geometry/metrics.py

```diff
@@ -391,6 +393,12 @@ def inverse_surface_metric_derivative(
     :arg xyz_axis: an integer denoting the physical coordinate axis.
     :arg dd: a :class:`~grudge.dof_desc.DOFDesc`, or a value convertible to one.
         Defaults to the base volume discretization.
+    :arg _use_geoderiv_connection: If *True*, some of the returned group arrays
```

Comment:
Perhaps we can give this a better name? I know mechanically it uses the new geoderivative connection, but how about calling something along the lines of `_affine_geometry` or `_use_affine_storage`?

Reply:
I think I'd prefer to keep the name.

First, it's internal-facing (cf. the leading underscore), so I don't mind if the name is technical. It's part of the docstring not because I'd like it to be used outside of where it is currently, but rather to describe what it does. I'll add a note to that effect.

Second, I'd like this flag to prescribe "do whatever the geoderiv thing does", instead of suggesting that "it has its own agenda."

What do you think?

Reply:
https://github.com/inducer/grudge/compare/b4fd08a4de60181c247c1ce2cb121ed7d299f26b..4c2b185e739099a2abab91d52bb6b8148a91524b shows these changes. I've also made the flag defined specifically in terms of the geoderiv connection instead of blathering about P^0 for affine. To compensate, I've added the geoderiv connection to the docs (also with a warning sign about being internal) to explain what it's for.

Reply:
@inducer This is a perfectly fine alternative for me!